### PR TITLE
memoryview is a context manager

### PR DESCRIPTION
**Pull request by Zac-HD**
[Since Python 3.2](https://docs.python.org/3/library/stdtypes.html#memoryview.release), `memoryview` has been usable as a context manager, so this PR adds the appropriate methods to let Mypy know.  Found while working on python-trio/trio#543.

### CODE REVIEW

File: stdlib/3/builtins.pyi [hunk 1]
```diff
@@ -495,6 +495,8 @@ class memoryview(Sized, Container[int]):
     ndim = ...  # type: int
 
     def __init__(self, obj: Union[bytes, bytearray, memoryview]) -> None: ...
+    def __enter__(self) -> memoryview: ...
+    def __exit__(self, exc_type, exc_value, exc_tb) -> None: ...
```
Comment (srittau) [inline stdlib/3/builtins.pyi]: Please annotate `__exit__` with:

```python
    def __exit__(self, exc_type: Optional[Type[BaseException]], exc_val: Optional[BaseException], exc_tb: Optional[TracebackType]) -> bool: ...
```

**Comment by Zac-HD** [comment]
Thanks - and done!

File: stdlib/3/builtins.pyi [hunk 2]
```diff
@@ -495,6 +495,8 @@ class memoryview(Sized, Container[int]):
     ndim = ...  # type: int
 
     def __init__(self, obj: Union[bytes, bytearray, memoryview]) -> None: ...
+    def __enter__(self) -> memoryview: ...
+    def __exit__(self, exc_type: Optional[BaseException], exc_val: Optional[BaseException], exc_tb: Optional[TracebackType]) -> bool: ...
```
Comment (srittau) [inline stdlib/3/builtins.pyi]: `exc_type` should be `Optional[Type[BaseException]]`.

Comment (Zac-HD) [reply stdlib/3/builtins.pyi]: Wow, somehow I managed to drop the `Type` while copy-pasting and then manually deleted the extra brackets.  Fixed now!

**Comment by srittau** [comment]
Thank you! I will merge when the CI finishes successfully.

**Comment by Zac-HD** [comment]
Hurray!  Thanks for the fast and helpful reviews :heart_eyes:

**Pull request merged**